s3-endpoints.go

```diff
@@ -32,6 +32,18 @@ var awsS3EndpointMap = map[string]awsS3Endpoint{
 		"s3.us-east-2.amazonaws.com",
 		"s3.dualstack.us-east-2.amazonaws.com",
 	},
+	"us-iso-east-1": {
+		"s3.us-iso-east-1.c2s.ic.gov",
+		"s3.dualstack.us-iso-east-1.c2s.ic.gov",
```

Comment:
```
λ ping s3.dualstack.us-iso-east-1.c2s.ic.gov
Ping request could not find host s3.dualstack.us-iso-east-1.c2s.ic.gov. Please check the name and try again.
```

We can't really have non-existing domains in here. 

Leave empty and add an error if attempting to use this region with dual stack.

Reply:
do you see similar result with `s3.us-iso-east-1.c2s.ic.gov` as well?

Reply:
Yes. Just run ping or nslookup yourself - it is a bit faster than waiting for me to do it ;)

Reply:
Thank you, I tested out the endpoints from the jumpbox (Amazon Workspace) for `us-iso-east-1` and `us-iso-west-1` & they were reachable. However, I didn't check them outside of the jumpbox (my mistake here).

Reply:
would you mind approving @klauspost since the suggestion to throw error/panic won't be needed in case the region selected is one of the isolated region.

Reply:
@vnzongzna Would you please explain. Are these not expected to be publicly reachable?

Reply:
Correct @klauspost 
isolated endpoints are only reachable from within the environment.
Got the reference from [here](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/model/Region.html) 
<img width="788" alt="image" src="https://github.com/user-attachments/assets/6c269bf0-389e-4d55-842c-fb16e1ff9427" />

Reply:
okie